File: python/48_self_powers.py
def self_powers() -> int:
    total_sum: int = sum([n ** n for n in range(1, 11)])
    
    for n in range(11, 1001):
        total_sum += (n ** n) % (10 ** 10)
    
    return total_sum % (10 ** 10)

if __name__ == "__main__":
    print(self_powers())

File: c#/_48_SelfPowers.cs
using System;
using System.Numerics;

public class _48_SelfPowers{
	public static long Euler48(){
		BigInteger sum = 0;
		const long Mod = 10_000_000_000;
		for (int i = 1; i <= 1000; i++){
			sum += BigInteger.ModPow(i, i, Mod);
			sum %= Mod;
		}
		return (long) sum;
	}
	
	public static void Main(){
		Console.WriteLine(Euler48());
	}
}
